applications/vanilla/models/class.categorymodel.php

```diff
@@ -3601,6 +3601,103 @@ private static function recalculateAggregateCounts() {
         self::instance()->clearCache();
     }
 
+    /**
+     * Recalculate  CountDiscussions and CountComments columns for category.
+     *
+     * @param int $categoryId Category ID to update
+     * @return void
+     */
+    public static function recalculateCategoryCounts(int $categoryId) {
+        $sql = Gdn::sql()
+            ->select('DiscussionID', 'count', 'DiscussionCount')
+            ->select('CountComments', 'sum', 'CountComments')
+            ->from('Discussion')
+            ->where('CategoryID', $categoryId);
+
+        $exclude = Gdn::config('Vanilla.Archive.Exclude');
+        if ($exclude) {
+            $archiveDate = Gdn::config('Vanilla.Archive.Date');
+            if ($archiveDate) {
+                $sql->where('d.DateLastComment >', $archiveDate);
+            }
+        }
+        $row = $sql->get()->firstRow(DATASET_TYPE_ARRAY);
+
+        $discussionCount = $row['DiscussionCount'] ?? 0;
+        $commentCount = $row['CountComments'] ?? 0;
+
+        Gdn::sql()
+            ->update('Category')
+            ->set('CountDiscussions', $discussionCount)
+            ->set('CountComments', $commentCount)
+            ->where('CategoryID', $categoryId)
+            ->put();
+    }
+
+    /**
+     * Recalculate  CountAllDiscussions count column for category.
+     *
+     * @param int $categoryId Category ID to update
+     * @return void
+     */
+    public static function recalculateCategoryCountsAll(int $categoryId) {
```

Comment:
Similar to the other recalculation function, could we rename this to something with a more descriptive name, like `recalculateAggregatePostCounts`? Anytime I see "recalculate" and "all" in a method name, I think it's going to recalculate *all* of something. I know the column names are "CountAllDiscussions" and "CountAllComments", but I think including "all" in this method name could be a little misleading.

Reply:
:-)